### DISCUSSION TITLE
[Feature] Custom tags for pics

### DISCUSSION THREAD

**Discussion by lorvent** [immich-app/immich#3518] [Feature Request]
### The feature

prologue :
I tried to stay away from _from that so called app and company_ (because i don't want to handover my pics to them) but after trying many and many apps...nothing matched my requirements.... each app lacks one or other feature and facial recognition is a complete miss in almost all apps including nextcloud, so finally (with no choice) i uploaded pics to that app.

today i was so amazed with this your app, trust me...i was so excited and happy seeing yours.

---
when using that app, it ticked many boxes except tag feature...
ex: when i go to restaurants, i take their menu pics but finding them after a year is not so easy so i prefer a tag feature and sometimes i may delete them in gallery so better to have it in cloud.

i searched for many such solutions but no luck,
finally settled with another OS app aaves but it stores only locally so there is a chance of missing those images when moving to new phone or accidentally deleting files.

so would be great, if you can add such feature.

ability to add custom tags to uploaded pictures.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by lefuturiste** [comment]
I agree, I also have the need to manually tags pictures. I mainly use immich to manage my large collection of street imagery and I have some recurring objects like defibrillator, fire hydrants, opening hours signs, street signs etc. I also have the need to tags with things that are specifics "theme" in my life, for example photos with my family, photos with my sport club, local monument photos. To be able to sort out different circles of my life.

**Comment by lefuturiste** [comment]
Similar to 
- https://github.com/immich-app/immich/discussions/1651
- https://github.com/immich-app/immich/discussions/2168

**Comment by JesseWebDotCom** [comment]
I also want to be able to add tags to people. Helps with the "who's that person" as well as quick one off searches like "who was that guy you worked with..."

**Reply by alextran1502** [reply]
what is the different of this vs using the name?

**Reply by JesseWebDotCom** [reply]
Tags ease searches by allowing multidimensional categorization in an unobtrusive way.

So let's say I want my son Jack to easily be able to find his friends.

- John Doe (soccer, baseball, pre-k)
- Mike Johnson (soccer, high school)
- Dan Lee (soccer, baseball, high school)
- Brian May (band, high school)
Etc

By modifying the person name, I'd have really long names like John Doe Jack Soccer Baseball Pre-K

With tags, the names stay clean / short (as they don't include this extra tag metadata) and I can search for tag combinations like jack+soccer+high school to only see his soccer friends from high school.

This is the same reasoning to why you add metadata to photos.

**Reply by lorvent** [reply]
in my case, i like to add tags like pass port, hdfc, cc, sbi, loan papers, land papers, city1, city2 etc.... i add 1 or more tags to each photo so i can search for any tag depending on purpose.

down the line in 6months... after adding 1000s of pics, it is very hard to remember date and scrolling through all pics to find what we wanted is also tough.

also adding albums for each tag i mentioned is also very bad idea.

**Comment by mglaske** [comment]
Agreed, I would also like this feature!

**Comment by Bobu5** [comment]
As a past samsung gallery user i couldn't agree more, definitely one of the features that has been holding me back from ditching samsung gallery and fully switch to Immich, but other than that Immich itself has been a great way to store all of my photos

**Comment by omarcinp** [comment]
I'm also looking for this feature!

**Comment by platima** [comment]
Same

**Comment by marciopamplona** [comment]
Related to #11450

**Comment by bo0tzz** [comment]
Will be closed by #11980.
